signer/tests/integration/postgres.rs

```diff
@@ -3061,6 +3063,815 @@ async fn deposit_report_with_deposit_request_confirmed() {
+            &qualified_id,
+            signer_public_key,
+        )
+        .await
+        .unwrap()
+        .unwrap();
+
+    assert_eq!(report.is_accepted, Some(true));
+    assert_eq!(report.status, WithdrawalRequestStatus::Confirmed);
+
+    // Let's try one more time but with another public key (who we know has
+    // not submitted a vote).
+    let signer_public_key_2 = &signer_public_keys[1];
+    let report = db
+        .get_withdrawal_request_report(
+            &bitcoin_chain_tip,
+            &stacks_chain_tip,
+            &qualified_id,
+            signer_public_key_2,
+        )
+        .await
+        .unwrap()
+        .unwrap();
+
+    assert!(report.is_accepted.is_none());
+    assert_eq!(report.status, WithdrawalRequestStatus::Confirmed);
+
+    testing::storage::drop_db(db).await;
+}
+
+/// Check that the report will return that the is unconfirmed if the
+/// transaction that generated the request is not on stacks blockchain
+/// identified by the given chain tip.
+#[tokio::test]
+async fn withdrawal_report_with_withdrawal_request_reorged() {
+    let db = testing::storage::new_test_database().await;
+    let mut rng = rand::rngs::StdRng::seed_from_u64(8);
+
+    // We only want the blockchain to be generated
+    let num_signers = 3;
+    let test_params = testing::storage::model::Params {
+        num_bitcoin_blocks: 10,
+        num_stacks_blocks_per_bitcoin_block: 1,
+        num_deposit_requests_per_block: 0,
+        num_withdraw_requests_per_block: 0,
+        num_signers_per_request: num_signers,
+        consecutive_blocks: false,
+    };
+
+    let signer_public_keys = testing::wsts::generate_signer_set_public_keys(&mut rng, num_signers);
+    let signer_public_key = &signer_public_keys[0];
+    let test_data = TestData::generate(&mut rng, &signer_public_keys, &test_params);
+    test_data.write_to(&db).await;
+
+    let bitcoin_chain_tip_ref = db
+        .get_bitcoin_canonical_chain_tip_ref()
+        .await
+        .unwrap()
+        .unwrap();
+    let bitcoin_chain_tip = bitcoin_chain_tip_ref.block_hash;
+    let stacks_chain_tip_block = db
+        .get_stacks_chain_tip(&bitcoin_chain_tip)
+        .await
+        .unwrap()
+        .unwrap();
+
+    // Okay let's put the withdrawal request in the database on the stacks
+    // chain tip.
+    let withdrawal_request = WithdrawalRequest {
+        block_hash: stacks_chain_tip_block.block_hash,
+        block_height: bitcoin_chain_tip_ref.block_height,
+        ..Faker.fake_with_rng(&mut rng)
+    };
+
+    db.write_withdrawal_request(&withdrawal_request)
+        .await
+        .unwrap();
+
+    // Now let's generate a report where we know that the withdrawal
+    // request is not on the blockchain identified by the given chain tip.
+    let qualified_id = withdrawal_request.qualified_id();
+    let report = db
+        .get_withdrawal_request_report(
+            &bitcoin_chain_tip,
+            &stacks_chain_tip_block.parent_hash,
```

Comment:
Could we use a random hash here or something that doesn't use the `stacks_chain_tip_block`? Took me a minute to see what was going on (maybe I'm just extra slow today) 😅

Reply:
😄 not a problem, I'll update it.

Reply:
Updated in https://github.com/stacks-network/sbtc/pull/1344/commits/dcfdfa3c4c1dd9c65bfd8d75f46b21c7dde54e88